sympy/printing/ccode.py

```diff
@@ -170,13 +170,33 @@ def _print_Rational(self, expr):
 
     def _print_Indexed(self, expr):
         # calculate index for 1d array
-        dims = expr.shape
-        elem = S.Zero
-        offset = S.One
-        for i in reversed(range(expr.rank)):
-            elem += expr.indices[i]*offset
-            offset *= dims[i]
-        return "%s[%s]" % (self._print(expr.base.label), self._print(elem))
+        offset = expr.offset
+        strides = expr.strides
+
+        if strides is None or isinstance(strides, str):
+            dims = expr.shape
+            elem = S.Zero
+            shift = S.One
+            if strides == 'C' or strides is None:
+                traversal = reversed(range(expr.rank))
+            elif strides == 'F':
+                traversal = range(expr.rank)
+
+            for i in traversal:
+                elem += expr.indices[i]*shift
+                shift *= dims[i]
+
+            if offset is None:
+                offset = ""
+            return "%s[%s]" % (self._print(expr.base.label),
```

Comment:
instead of having a separate return statement we can calculate the missing `strides` here and let the method use the same logic further down when strides is given explicitly.